hw2/Naetw/q2/pybind_test.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+from _vector import angle
+import math
+import numpy as np
+import pytest
+
+class TestClass:
+    m_abs_tol = 1e-05
+
+    def test_invalid(self):
+        vector1 = np.random.randint(64, 1922, 2)
+        vector2 = np.array([0, 0])
+        vector3 = np.random.randint(64, 1922, 2)
+
```

Comment:
how about infinity, non-number arguments, 0.0000000001?

Reply:
It's the caller that is responsible for a non-compatible argument (non-number argument), right? This test is for testing the functionality of `angle()` which is, in this situation, a callee. I don't think I have to test for a non-compatible argument.

For infinity, I can test it whether it's NaN or not.

As for 0.0000000001, I don't get what's your point.

Reply:
Yes, I agree with testing only for number arguments.

umm, because your precision is `float`, so I just wonder what happens if the number below to the precision.

Reply:
The range of single-precision floating-point is 1.2E-38 to 3.4E+38. I think 0.0000000001 (1e-10) is within the range of it. As for the number that is off the single-precision ability, it's like passing a `double` to a `float`. It may lose some bits. And I think it's the same issue as what we have in testing for the non-compatible argument.